### PR TITLE
Added convergence tester to four roll mill examples.

### PR DESCRIPTION
**Pull request by abarret**
@bindi-nagda I've added a target `frm_convergence` target that should compute difference norms between two successive grids. It expects the ratio between coarse run and fine run to be 2 (i.e. `N=32` and `N=64`). It also will only run on a single time step.

### CODE REVIEW

File: frm/input2d.convergence [hunk 1]
```diff
@@ -0,0 +1,39 @@
+// grid spacing parameters
+L = 1.0
+MAX_LEVELS = 1  // maximum number of levels in locally refined grid
+REF_RATIO  = 2  // refinement ratio between levels
+N = 32          // coarsest grid spacing (Assumes equal number of grid points in each direction)
```
Comment (abarret) [inline frm/input2d.convergence:5]: These should refer to the coarse simulation.

File: frm/input2d.convergence [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+Main {
+
+coarse_hier_dump_dirname = "hier_data_IB2d_coarse"
+fine_hier_dump_dirname = "hier_data_IB2d_fine"
```
Comment (abarret) [inline frm/input2d.convergence:10]: You should specify the directory in which the data was dumped for each run.

File: frm/input2d.convergence [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+coarse_iter_num = 1000
+fine_iter_num = 2000
```
Comment (abarret) [inline frm/input2d.convergence:13]: These should be the time steps number that you want to compare.

File: frm/four_roll_mill.cpp [hunk 4]
```diff
@@ -171,6 +184,46 @@ main(int argc, char* argv[])
...
 } // main
+
+void
+output_data(Pointer<PatchHierarchy<NDIM> > patch_hierarchy,
+                 Pointer<INSVCTwoFluidStaggeredHierarchyIntegrator> ins_integrator,
+                 Pointer<AdvDiffSemiImplicitHierarchyIntegrator> adv_diff_integrator,
+                 const int iteration_num,
+                 const double loop_time,
+                 const string& data_dump_dirname)
+{
```
Comment (bindi-nagda) [inline frm/four_roll_mill.cpp:205]: Added this function that creates a hierarchy data dump file.

Comment (bindi-nagda) [inline frm/four_roll_mill.cpp:205]: Added this function that creates a hierarchy data dump file.

File: frm/four_roll_mill.cpp [hunk 5]
```diff
@@ -171,6 +184,46 @@ main(int argc, char* argv[])
                 visit_data_writer->writePlotData(patch_hierarchy, iteration_num, loop_time);
                 next_viz_dump_time += viz_dump_time_interval;
             }
+
+            // At specified intervals, store hierarchy data for post processing.
+            const bool last_step = !ins_integrator->stepsRemaining();
+            if (dump_postproc_data && last_step)   
+            {
+                output_data(patch_hierarchy, ins_integrator, adv_diff_integrator, iteration_num, loop_time, postproc_data_dump_dirname);
+            }
```
Comment (abarret) [inline frm/four_roll_mill.cpp:193]: You can just put this after the `while` loop.

File: frm/four_roll_mill.cpp (hunk 4 above)
Comment (abarret) [reply frm/four_roll_mill.cpp:205]: :+1:

**Comment by abarret** [comment]
Once @bindi-nagda approves this, I will merge.

**Review by bindi-nagda — APPROVED** [review]

**Pull request merged**